unit/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -669,3 +669,230 @@ TEST_CASE(
+  const auto original_array_symbol =
+    make_test_symbol("original_array", array_type);
+  const auto result_array_symbol = make_test_symbol("result_array", array_type);
+  with_exprt with_expr{
+    original_array_symbol.symbol_expr(),
+    from_integer(0, index_type),
+    from_integer(0, value_type)};
+  with_expr.add_to_operands(
+    from_integer(1, index_type), from_integer(1, value_type));
+  with_expr.add_to_operands(
+    from_integer(2, index_type), from_integer(2, value_type));
+  const equal_exprt equal_expr{result_array_symbol.symbol_expr(), with_expr};
+  test.sent_commands.clear();
+  test.procedure.set_to(equal_expr, true);
+
+  const smt_bit_vector_sortt smt_index_sort{32};
+  const smt_bit_vector_sortt smt_value_sort{8};
+  const smt_array_sortt smt_array_sort{smt_index_sort, smt_value_sort};
+  const smt_identifier_termt smt_original_array_term{
+    "original_array", smt_array_sort};
+  const smt_identifier_termt smt_result_array_term{
+    "result_array", smt_array_sort};
+  const smt_identifier_termt index_0_term{"index_0", smt_index_sort};
+  const smt_identifier_termt index_1_term{"index_1", smt_index_sort};
+  const smt_identifier_termt index_2_term{"index_2", smt_index_sort};
+  const smt_termt store_term = smt_array_theoryt::store(
+    smt_array_theoryt::store(
+      smt_array_theoryt::store(
+        smt_original_array_term,
+        index_0_term,
+        smt_bit_vector_constant_termt{0, smt_value_sort}),
+      index_1_term,
+      smt_bit_vector_constant_termt{1, smt_value_sort}),
+    index_2_term,
+    smt_bit_vector_constant_termt{2, smt_value_sort});
+  const auto smt_assertion = smt_assert_commandt{
+    smt_core_theoryt::equal(smt_result_array_term, store_term)};
+  const std::vector<smt_commandt> expected_commands{
+    smt_declare_function_commandt(smt_result_array_term, {}),
+    smt_declare_function_commandt(smt_original_array_term, {}),
+    smt_define_function_commandt{
+      index_0_term.identifier(),
+      {},
+      smt_bit_vector_constant_termt{0, smt_index_sort}},
+    smt_define_function_commandt{
+      index_1_term.identifier(),
+      {},
+      smt_bit_vector_constant_termt{1, smt_index_sort}},
+    smt_define_function_commandt{
+      index_2_term.identifier(),
+      {},
+      smt_bit_vector_constant_termt{2, smt_index_sort}},
+    smt_assertion};
+  REQUIRE(test.sent_commands == expected_commands);
+}
+
+TEST_CASE(
+  "smt2_incremental_decision_proceduret byte update-extract commands.",
+  "[core][smt2_incremental]")
+{
+  auto test = decision_procedure_test_environmentt::make();
+  SECTION("byte_extract - byte from int correctly extracted.")
+  {
+    const auto int64_type = signedbv_typet(64);
+    const auto byte_type = signedbv_typet(8);
+    const auto extracted_byte_symbol =
+      make_test_symbol("extracted_byte", byte_type);
+    const auto original_int_symbol =
+      make_test_symbol("original_int", int64_type);
+    const auto byte_extract = make_byte_extract(
```

Comment:
:exclamation: This commit does not compile when checked out, due to a missing `#include <util/byte_operators.h>` needed for this line in the unit test.

Reply:
I think the problems with this particular test are just down to it being squashed into the wrong commit.